### ISSUE TITLE
False error 'UnusedThrows' when using new SplFileObject

### ISSUE THREAD

**Issue by VincentLanglet** [pepakriz/phpstan-exception-rules#106]
The code
```
	/**
	 * @throws \Exception
	 */
	public function shouldBeCorrect()
	{
		new \SplFileObject(''); // The phpdoc say @throws RuntimeException
	}
```
Is failing with the error message ` Unused @throws Exception annotation`

See https://github.com/pepakriz/phpstan-exception-rules/pull/105

**Comment by pepakriz** [comment]
@VincentLanglet You can write a `ConstructorThrowTypeExcention` to handle it correctly. See a similar solution for `DateTime`: https://github.com/pepakriz/phpstan-exception-rules/blob/master/src/Extension/DateTimeExtension.php

**Comment by VincentLanglet** [comment]
@pepakriz Indeed I'll have to write this if I want to handle dynamically the exception.

But the phpdoc say clearly there can be an exception
```
 /**
         * Construct a new file object.
         *
         * @link  https://php.net/manual/en/splfileobject.construct.php
         *
         * @param string   $file_name        The file to open
         * @param string   $open_mode        [optional] The mode in which to open the file. See {@see fopen} for a list of allowed modes.
         * @param bool     $use_include_path [optional] Whether to search in the include_path for filename
         * @param resource $context          [optional] A valid context resource created with {@see stream_context_create}
         *
         * @throws RuntimeException When the filename cannot be opened
         * @throws LogicException When the filename is a directory
         *
         * @since 5.1
         */
        public function __construct ($file_name, $open_mode = 'r', $use_include_path = false, $context = null) {}
```

So why, by default I don't have to catch the exception since there is no ConstructorThrowTypeExcention ? Why is there special case for constructor ?

**Comment by pepakriz** [comment]
@VincentLanglet It's stub file from your IDE, right? Internal objects don't have phpdoc.

**Comment by VincentLanglet** [comment]
Yes you're right. I didn't know that.

**Comment by pepakriz** [comment]
@VincentLanglet I'll keep this issue opened. Do you want to try to make a PR for `SplFileObjectExtension`? I'd like to release next version with this feature.

**Comment by VincentLanglet** [comment]
I'll see if I can
